lightning/src/routing/gossip.rs

```diff
@@ -318,11 +318,10 @@ where C::Target: chain::Access, L::Target: Logger
 		Ok(msg.contents.excess_data.len() <= MAX_EXCESS_BYTES_FOR_RELAY)
 	}
 
-	fn get_next_channel_announcements(&self, starting_point: u64, batch_amount: u8) -> Vec<(ChannelAnnouncement, Option<ChannelUpdate>, Option<ChannelUpdate>)> {
-		let mut result = Vec::with_capacity(batch_amount as usize);
+	fn get_next_channel_announcement(&self, starting_point: u64) -> Option<(ChannelAnnouncement, Option<ChannelUpdate>, Option<ChannelUpdate>)> {
 		let channels = self.network_graph.channels.read().unwrap();
 		let mut iter = channels.range(starting_point..);
-		while result.len() < batch_amount as usize {
+		loop {
 			if let Some((_, ref chan)) = iter.next() {
```

Comment:
Maybe just make this a `for` loop instead of explicitly manipulating the iterator?